design/survival/design_survival.qmd

```diff
@@ -0,0 +1,1825 @@
+---
+title: "Design Doc for Survival Analysis with RobinCar2"
+author: Daniel Sabanes Bove
+date: "`r format(Sys.Date(), '%d %b %Y')`"
+date-format: long
+bibliography: references.bib
+number-sections: true
+---
+
+This design doc shall serve to align on the design of the survival analysis implementation in RobinCar2.
+
+# User interface
+
+The proposal for the user interface is the function call of the form:
+
+```{r}
+#| eval: false
+
+robin_surv(
+  Surv(time, event) ~ treatment * s1 + covariates,
+  data = df,
+  treatment = pb(s1),
+  contrast = "hazardratio",
+  pval = "logrank"
```

Comment:
One alternative to `pval` would be to name this argument `statistic`, which could emphasize that it's a different test statistic. When I see `pval` my first thought is "is it one or two sided" rather than "what test statistic to use"